File: App/Resources/AlgorithmDetails/stoogesort/stoogesort.go
package main

import (
	"fmt"
)

func stoogeSort(arr []int, i, j int) {
	if arr[j] < arr[i] {
		arr[i], arr[j] = arr[j], arr[i]
	}
	if j-i > 1 {
		t := (j - i + 1) / 3
		stoogeSort(arr, i, j-t)
		stoogeSort(arr, i+t, j)
		stoogeSort(arr, i, j-t)
	}
}

func sort(arr []int) []int {
	stoogeSort(arr, 0, len(arr)-1)
	return arr
}

func main() {
	array := []int{0, 39, 21, 62, 91, 77, 14, 23,
		90, 69, 51, 81, 68, 83, 32, 56}
	fmt.Println(sort(array))
}

File: App/Resources/AlgorithmDetails/stoogesort/stoogesort.cs
using System;

public class StoogeSort
{
  public static void Sort(int[] arr, int i, int j)
  {
    if (arr[j].CompareTo(arr[i]) < 0)
    {
      (arr[i], arr[j]) = (arr[j], arr[i]);
    }
    if (j - i > 1)
    {
      int t = (j - i + 1) / 3;
      Sort(arr, i, j - t);
      Sort(arr, i + t, j);
      Sort(arr, i, j - t);
    }
  }

  public static void Main(String[] args)
  {
    int[] array = { 0, 39, 21, 62, 91, 77, 14, 23, 90, 69, 51, 81, 68, 83, 32, 56 };
    Sort(array, 0, array.Length - 1);
    string result = "[" + String.Join(", ", array) + "]";
    Console.WriteLine(result);
  }
}
